Analysis/src/SignalObjectSelector.cpp

```diff
@@ -38,24 +38,100 @@ SignalObjectSelector::SignalObjectSelector(SignalMode _mode) : mode(_mode)
+        { {Channel::ETau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::Tight} },
+        { {Channel::MuTau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::TauTau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::ETau, SignalMode::HTT}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::Tight} },
+        { {Channel::MuTau, SignalMode::HTT}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::TauTau, SignalMode::HTT}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::ETau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::MuTau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::TauTau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::ETau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::MuTau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::TauTau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VVVLoose} },
+        { {Channel::ETau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} },
+        { {Channel::MuTau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} },
+        { {Channel::TauTau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} },
+        { {Channel::ETau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::Tight} },
+        { {Channel::MuTau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::TauTau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstElectronMVA6, DiscriminatorWP::VLoose} },
+        { {Channel::ETau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} },
+        { {Channel::MuTau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} },
+        { {Channel::TauTau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::VVVLoose} }
+        // { {Channel::ETau, SignalMode::Skimmer}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::Tight} },
+        // { {Channel::ETau, SignalMode::TauPOG_Skimmer}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::Tight} },
+    };
+    std::pair<TauIdDiscriminator, DiscriminatorWP> disc_wp;
```

Comment:
it is a map, don't need to loop. use `tauVSeDiscriminators.find`. If not found -> throw an exception